In [1]:
# Import packages 
import warnings
warnings.filterwarnings('ignore')
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import numpy as np
from sklearn.linear_model import LinearRegression
sns.set()

# Set global options 
pd.options.display.max_rows = 50
pd.options.display.max_columns = 100
plt.style.use("seaborn-whitegrid")
sns.set_color_codes("muted")

In [2]:
df= pd.read_excel('dataset_giaxemaycu.xlsx')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 219 entries, 0 to 218
Data columns (total 17 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Hãng xe              219 non-null    object
 1   Trọng lượng          219 non-null    object
 2   Mã lực               219 non-null    object
 3   Màu xe               219 non-null    object
 4   Năm sản xuất         219 non-null    int64 
 5   Dòng xe              219 non-null    object
 6   Loại xe              219 non-null    object
 7   Tên xe               219 non-null    object
 8   Giá bán              219 non-null    int64 
 9   Năm đăng ký          219 non-null    int64 
 10  Tình trạng           219 non-null    object
 11  Dung Tích            219 non-null    object
 12  Xuất xứ              219 non-null    object
 13  Số km đã đi          219 non-null    object
 14  Chính sách bảo hành  219 non-null    object
 15  Giấy tờ            219 non-null    object
 16  Địa chỉ 

## Tổng quan dữ liệu

In [4]:
# Xem số dòng, số cột
df.shape

(219, 17)

In [5]:
# Xem một số dòng đầu 
df.head()

Hãng xe Trọng lượng Mã lực  Màu xe  Năm sản xuất    Dòng xe  Loại xe  \
0  Yamaha         117   15.4     Đỏ           2018    Exciter  Côn tay   
1   Honda         110   11.6   Xanh           2018  Air Blade   Tay ga   
2  Yamaha          99    8.8     Đen          2018     Sirius    Xe số   
3  Yamaha          99    8.1   Xanh           2016     Grande   Tay ga   
4   Honda          97    8.8  Trắng          2020     Vision   Tay ga   

                      Tên xe   Giá bán  Năm đăng ký Tình trạng Dung Tích  \
0        Yamaha Exciter 2018  16800000         2018        0.7       150   
1       Honda Air Blade 2018  26500000         2018        0.6       125   
2         Yamaha Sirius 2018   9800000         2018        0.6       115   
3         Yamaha Grande 2016  19500000         2016        0.6       125   
4  Honda Vision Smarkey 2020  25500000         2020       0.85       110   

    Xuất xứ Số km đã đi Chính sách bảo hành  Giấy tờ Địa chỉ người bán  
0  Việt Nam       30000                Hãng  Chính chủ        Bình Dương  
1  Nhật Bản        7979                Hãng  Chính chủ           Đắk Lắk  
2  Việt Nam       20000                Hãng  Chính chủ           Cần Thơ  
3  Việt Nam       20000                Hãng  Chính chủ            Hà Nội  
4  Việt Nam       10000                Hãng  Chính chủ            Hà Nội

In [6]:
# Xem một số dòng cuối để chắc rằng dữ liệu là đồng nhấu 
df.tail(4)

Hãng xe Trọng lượng Mã lực Màu xe  Năm sản xuất   Dòng xe  Loại xe  \
215    Yamaha         116   15.4    Xám          2022       NVX   Tay ga   
216     Honda         130   14.4  Vàng           2015      PCX   Tay ga    
217     Honda          96    8.8    Đỏ           2022    Vision   Tay ga   
218  Piaggio          116   11.7  Trắng          2015  Liberty   Tay ga    

                   Tên xe   Giá bán  Năm đăng ký Tình trạng Dung Tích  \
215        Yamaha NVX 155  38800000         2022        0.8       155   
216         Honda PCX 150  21500000         2015        0.6       157   
217     Honda Vision 2022  30500000         2022       0.85       110   
218  Piaggio Liberty 2015   5800000         2015        0.8     124.5   

       Xuất xứ Số km đã đi Chính sách bảo hành         Giấy tờ  \
215   Nhật Bản        2300                Hãng         Chính chủ   
216  Việt Nam        76000       Hết thời hạn         Chính chủ    
217   Nhật Bản        4000                Hãng         Chính chủ   
218  Việt Nam        20000       Hết thời hạn   Không chính chủ    

    Địa chỉ người bán  
215            Hà Nội  
216          Đà Nẵng   
217           Đà Nẵng  
218           Hà Nội

In [7]:
df.describe()

Năm sản xuất       Giá bán  Năm đăng ký
count    219.000000  2.190000e+02   219.000000
mean    2017.885845  3.065985e+07  2017.972603
std        4.346329  2.912744e+07     4.356707
min     1987.000000  1.700000e+06  1987.000000
25%     2016.000000  1.485000e+07  2016.000000
50%     2019.000000  2.350000e+07  2019.000000
75%     2021.000000  3.600000e+07  2021.000000
max     2023.000000  2.990000e+08  2023.000000

** COMMENT **
- Dataset về xe máy cũ
- Gồm 219 rows và 17 columns
- 

# 1. Tiền xử lý dữ liệu

#### Chuẩn hóa tên cột 
    - Viết thường
    - Thay thế khoảng trắng bằng dấu _

In [8]:
# View column's names
df.columns.to_list()

['Hãng xe',
 'Trọng lượng',
 'Mã lực',
 'Màu xe',
 'Năm sản xuất',
 'Dòng xe',
 'Loại xe',
 'Tên xe',
 'Giá bán',
 'Năm đăng ký',
 'Tình trạng',
 'Dung Tích',
 'Xuất xứ',
 'Số km đã đi',
 'Chính sách bảo hành',
 'Giấy tờ',
 'Địa chỉ người bán']

In [9]:
import re

#Standardize column names 
df.columns = [re.sub("[ -]", "_", c).lower().strip() for c in df.columns]

In [10]:
# Check result
df.head()

hãng_xe trọng_lượng mã_lực  màu_xe  năm_sản_xuất    dòng_xe  loại_xe  \
0  Yamaha         117   15.4     Đỏ           2018    Exciter  Côn tay   
1   Honda         110   11.6   Xanh           2018  Air Blade   Tay ga   
2  Yamaha          99    8.8     Đen          2018     Sirius    Xe số   
3  Yamaha          99    8.1   Xanh           2016     Grande   Tay ga   
4   Honda          97    8.8  Trắng          2020     Vision   Tay ga   

                      tên_xe   giá_bán  năm_đăng_ký tình_trạng dung_tích  \
0        Yamaha Exciter 2018  16800000         2018        0.7       150   
1       Honda Air Blade 2018  26500000         2018        0.6       125   
2         Yamaha Sirius 2018   9800000         2018        0.6       115   
3         Yamaha Grande 2016  19500000         2016        0.6       125   
4  Honda Vision Smarkey 2020  25500000         2020       0.85       110   

    xuất_xứ số_km_đã_đi chính_sách_bảo_hành  giấy_tờ địa_chỉ_người_bán  
0  Việt Nam       30000                Hãng  Chính chủ        Bình Dương  
1  Nhật Bản        7979                Hãng  Chính chủ           Đắk Lắk  
2  Việt Nam       20000                Hãng  Chính chủ           Cần Thơ  
3  Việt Nam       20000                Hãng  Chính chủ            Hà Nội  
4  Việt Nam       10000                Hãng  Chính chủ            Hà Nội

In [11]:
# Chuẩn hóa lại cột giá bán để dễ đọc
df['giá_bán'] = df['giá_bán'] / 1000000  # Chia cho 1 000 000 để đổi đơn vị ( ví dụ 16 800 000 thành 16.8)

In [16]:
#Check
df.head()

hãng_xe trọng_lượng mã_lực  màu_xe  năm_sản_xuất    dòng_xe  loại_xe  \
0  Yamaha         117   15.4     Đỏ           2018    Exciter  Côn tay   
1   Honda         110   11.6   Xanh           2018  Air Blade   Tay ga   
2  Yamaha          99    8.8     Đen          2018     Sirius    Xe số   
3  Yamaha          99    8.1   Xanh           2016     Grande   Tay ga   
4   Honda          97    8.8  Trắng          2020     Vision   Tay ga   

                      tên_xe  giá_bán  năm_đăng_ký tình_trạng dung_tích  \
0        Yamaha Exciter 2018     16.8         2018        0.7       150   
1       Honda Air Blade 2018     26.5         2018        0.6       125   
2         Yamaha Sirius 2018      9.8         2018        0.6       115   
3         Yamaha Grande 2016     19.5         2016        0.6       125   
4  Honda Vision Smarkey 2020     25.5         2020       0.85       110   

    xuất_xứ số_km_đã_đi chính_sách_bảo_hành  giấy_tờ địa_chỉ_người_bán  
0  Việt Nam       30000                Hãng  Chính chủ        Bình Dương  
1  Nhật Bản        7979                Hãng  Chính chủ           Đắk Lắk  
2  Việt Nam       20000                Hãng  Chính chủ           Cần Thơ  
3  Việt Nam       20000                Hãng  Chính chủ            Hà Nội  
4  Việt Nam       10000                Hãng  Chính chủ            Hà Nội

#### Check data quality

In [27]:
# Checking missing data


In [ ]:
# Check data types


# 2. Thống kê mô tả
    * summaries of important locations
    - min, max
    - mean, median, mode
    - quantile, percentile
    * sumaries of variation: varience, standard deviation
    * other: sum, kurtosis, skewness

# 3. EDA và Visualization